### ISSUE TITLE
Version 1.11 of Terraform kubernetes provider causes stack launch failure

### ISSUE THREAD

**Issue by darren-reddick** [kbst/terraform-kubestack#90]
Using the latest version of the kubernetes provider fails to launch the aws quickstart stack (quickstart/src/configurations/eks) with error:

> Error: Failed to initialize config: invalid configuration: no configuration has been provided

This seems to be a known issue - see https://github.com/terraform-providers/terraform-provider-kubernetes/issues/759

Pinning the provider at 1.10 seems to be a workaround for the issue:

```
quickstart/src/configurations/eks/versions.tf
terraform {
  required_version = ">= 0.12"
  required_providers {
    kubernetes = "1.10"
  }
}
```

**Comment by pst** [comment]
The provider versions are supposed to be pinned, see: https://github.com/kbst/terraform-kubestack/blob/master/aws/cluster/providers.tf 

Each cluster module has this.

If this does not ensure the provider has the correct version, this needs investigation. Moving the provider version pinning to the versions.tf may be one option.

Reading the issue you linked, it seems this does not only apply to EKS, but also GKE and AKS. It seems it's related to creating the cluster and the k8s resources on it in the same TF apply run. This is something Kubestack depends heavily on and which worked fine before this upstream change.

**Comment by darren-reddick** [comment]
I will try to look at this again later to see if I get the same behaviour. Although I did follow the quick start guide it is possible that I missed something.

**Comment by darren-reddick** [comment]
I tried this again and got the same behaviour.

After a bit of digging I found that it seems to be the way that the pessimistic constraint operator evaluates the allowed versions. If you include the patch version in the version string it seems to work as intended. e.g. 1.10.0 or alternatively use the exact version operator.

Someone has raised an issue here https://github.com/hashicorp/go-version/issues/55 some time ago but have no response.
